framework/src/meshgenerators/ExtraNodesetGenerator.C

```diff
@@ -132,13 +134,43 @@ ExtraNodesetGenerator::generate()
       this->comm().max(on_node);
     }
 
+    // only search for closest node if it is not found on a node
+    if (use_closest_node && !on_node)
+    {
+      // these are always true when using closest node
+      found_elem = true;
+      on_node = true;
+      Real dmin(std::numeric_limits<Real>::max());
+      dof_id_type dmin_id(std::numeric_limits<dof_id_type>::max());
+      // find this processors closest node and save distance and the node id
+      for (const auto & node_iter : as_range(mesh->local_nodes_begin(), mesh->local_nodes_end()))
+      {
+        Real dist = (*node_iter - p).norm_sq();
+        if (dist < dmin)
+        {
+          dmin = dist;
+          dmin_id = node_iter->id();
+        }
+      }
+
+      // get proc id with the global closest node
+      processor_id_type dmin_proc_id;
+      this->comm().minloc(dmin, dmin_proc_id);
+      // To make this work for distributed, only add on the proc_id containing the closest node
+      if (mesh->processor_id() == dmin_proc_id)
+        for (const auto & boundary_id : boundary_ids)
+          boundary_info.add_node(mesh->node_ptr(dmin_id), boundary_id);
```

Comment:
could probably use the mesh point locator here to do this
for next time.

Reply:
I was going to but that would limit me to only finding the node within an element that was closest to the point so that would require the point to be inside the domain.

Reply:
Yeah, sadly I wouldn't expect that to work in general.  If the mesh is anisotropic then the node closest to a point can be arbitrarily many layers of elements away from the element that contains that point, and if the mesh is self-overlapping and disconnected then the node closest to a point can be completely inaccessible from the element that contains that point.

IIRC libMesh has some k-D tree code that's not element-centric and can give accurate results for a case like this, but it wouldn't be an improvement in speed unless you were querying many many points after constructing the tree.